Adds craftable paper sacks

TehZombehz opened the pull request:
Adds paper sacks to the game, craftable by tablecrafting with 5 pieces of paper.
Comes in various designs. Use a pen on a paper sack to open a dialogue to change it's appearance [presumably using the pen's various color settings].
![bags](https://cloud.githubusercontent.com/assets/15373249/14071246/bf4e1e80-f477-11e5-8ff9-5ac56d555d85.png)
~~I totally plan to add the ability to cut holes into one and wear them as a hat. Soon(TM)~~
You can use a sharp object on the regular and smiley face versions of paper sacks to craft their respective paper sack hat versions. Perfect for hiding your shame or looking really creepy.
![sackhat](https://cloud.githubusercontent.com/assets/15373249/14073194/abed80d2-f48c-11e5-996e-36786c6885f6.png)
:cl:
rscadd: Paper sacks are now craftable by tablecrafting with 5 sheets of paper.
rscadd: Comes in 5 different designs: Use a pen on a paper sack to change the design.
rscadd: Certain paper sack designs can be modified with a sharp object to craft a (creepy) paper sack hat.
/:cl:

TehZombehz committed:
Adds craftable paper sacks

Adds craftable paper sacks

Bawhoppen commented:
what a nice and pleasant addition to the game

it doesn't fit in cold dead spessmens world

TehZombehz committed:
Requirement to hold paper bag to change design removed

KazeEspada commented:
Doesn't fit in head slot....

TrustyGun commented:
>I totally plan to add the ability to cut holes into one and wear them as a hat. Soon(TM)

PKPenguin321 commented:
i envy your spriting abilities

TehZombehz committed:
Regular and smiley paper sack are now wearable if modified

Use a sharp object on an empty regular or smiley face paper sack to
craft their respective wearable paper sack hats.

TehZombehz committed:
Typo fix

TehZombehz committed:
More typo fixing

I swear I'm good at english

Shapsy commented:
Adorable :+1:

Iamgoofball commented:
antag type that steals the souls of anyone who removes the bag when?

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+			if(2)
+				icon_state = "paperbag_syndi"
+				item_state = "paperbag_syndi"
+				desc = "The design on this paper sack is a remnant of the secrative 'SydiSnacks' program."
```
secretive

PKPenguin321 replied:
also "SyndieSnacks" not "SydiSnacks"

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+			if(1)
+				icon_state = "paperbag_nt"
+				item_state = "paperbag_nt"
+				desc = "A Nanotrasen standard paper lunch sack for loyal employees on the go."
```
standard Nanotrasen would probably make more sense but that is a nitpick

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+	foldable = null
+	var/design = 0 /* 0 = No design, 1 = Nanotrasen Standard, 2 = Syndisnacks, 3 = Heart, 4 = Smiley Face */
+
+/obj/item/weapon/storage/box/papersack/update_icon()
```
Is this already called whenever you put things into a storage item?

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+		if(get_dist(usr, src) > 1)
+			usr << "<span class='warning'>You have moved too far away!</span>"
+			return
+		design = designs.Find(switchDesign) - 1
```
this doesn't look like the usual byond coding to me

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+	item_state = "paperbag"
+	burn_state = FLAMMABLE
+	foldable = null
+	var/design = 0 /* 0 = No design, 1 = Nanotrasen Standard, 2 = Syndisnacks, 3 = Heart, 4 = Smiley Face */
```
use defines instead of weird numbers you have to comment on

#define NANOTRASEN "Nanotrasen Standard" etc

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+		if(contents.len)
+			user << "<span class='warning'>You can't modify this [src] with items still inside!</span>"
+			return
+		var/list/designs = list("None", "Nanotrasen Standard", "SyndiSnacks", "Heart", "Smiley Face", "Cancel")
```
And then you can list the defines here

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+		if(design < 5)
+			usr << "<span class='notice'>You make some modifications to the [src] using your pen.</span>"
+		switch(design)
+			if(0)
```
And use the defines here

KorPhaeron commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,67 @@
+			usr << "<span class='notice'>You make some modifications to the [src] using your pen.</span>"
+		switch(design)
+			if(0)
+				icon_state = "paperbag"
```
you can probably do icon_state = paperbag[design] once instead of inside each if

TehZombehz committed:
Changes Kor requested #1

TehZombehz committed:
Type replacement

WJohn commented:
:+1: on the sprites.

Chocobro commented:
:+1:
also @TehZombehz requesting a :3

phil235 commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -593,3 +594,73 @@
+				new /obj/item/clothing/head/papersack(user.loc)
+				qdel(src)
+				return 0
+			else if(item_state == "paperbag_SmileyFace")
```
I don't see anywhere in the code where item_state becomes "paperbag_SmileyFace"
